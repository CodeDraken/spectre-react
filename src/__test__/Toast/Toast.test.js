import React from 'react'
import { shallow } from 'enzyme'

import { Toast } from '../..'

describe('Toasts', () => {
  it('Should render a div tag by default', () => {
    const wrapper = shallow(<Toast />)

    expect(wrapper.type()).toBe('div')
  })

  it('Should render children', () => {
    const wrapper = shallow(<Toast>My toast</Toast>)

    expect(wrapper.text()).toBe('My toast')
  })

  it('Should pass additional classNames', () => {
    const wrapper = shallow(<Toast className="extra">My Toasts</Toast>)

    expect(wrapper.hasClass('extra')).toBe(true)
  })

  it('Should render with "Toasts" class by default', () => {
    const wrapper = shallow(<Toast>Toasts</Toast>)

    expect(wrapper.hasClass('toast')).toBe(true)
  })

  it('Should render with "toast-{color}" class when a color is provided', () => {
    const wrapper = shallow(<Toast color="primary">Primary</Toast>)

    expect(wrapper.hasClass('toast-primary')).toBe(true)
  })

  it('Should render custom tag', () => {
    const wrapper = shallow(<Toast renderAs="main">custom</Toast>)

    expect(wrapper.type()).toBe('main')
  })

  it('Should match the snapshot', () => {
    const wrapper = shallow(<Toast />)

    expect(wrapper).toMatchSnapshot()
  })

  // it('Should have toast classname', () => {
  //   const wrapper = renderer.create(<Toast>Test</Toast>)

  //   expect(wrapper.toJSON()).toMatchSnapshot()
  // })

  // it('Should concat classnames in props with Spectre classnames', () => {
  //   const wrapper = renderer.create(
  //     <Toast className='other-class this-is-a-test'>classes</Toast>
  //   )

  //   expect(wrapper.toJSON()).toMatchSnapshot()
  // })

  // it('Should use inline styles', () => {
  //   const wrapper = renderer.create(
  //     <Toast style={{ width: 250 }}>Inline styles</Toast>)

  //   expect(wrapper.toJSON()).toMatchSnapshot()
  // })
})
