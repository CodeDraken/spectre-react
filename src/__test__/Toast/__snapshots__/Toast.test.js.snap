// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Toasts Should match the snapshot 1`] = `
<div
  className="toast"
/>
`;
